src/java/org/apache/cassandra/service/MigrationTask.java

```diff
@@ -65,30 +76,51 @@ public void runMayThrow() throws Exception
 
         MessageOut message = new MessageOut<>(MessagingService.Verb.MIGRATION_REQUEST, null, MigrationManager.MigrationsSerializer.instance);
 
-        IAsyncCallback<Collection<Mutation>> cb = new IAsyncCallback<Collection<Mutation>>()
+        IAsyncCallbackWithFailure<Collection<Mutation>> cb = new IAsyncCallbackWithFailure<Collection<Mutation>>()
         {
             @Override
             public void response(MessageIn<Collection<Mutation>> message)
             {
                 try
                 {
-                    LegacySchemaTables.mergeSchema(message.payload);
+                    // TODO: testing always failed merges
+                    //LegacySchemaTables.mergeSchema(message.payload);
                 }
+                /*
                 catch (IOException e)
                 {
                     logger.error("IOException merging remote schema", e);
                 }
+                */
                 catch (ConfigurationException e)
                 {
                     logger.error("Configuration exception merging remote schema", e);
                 }
+                finally
+                {
+                    // always attempt to clean up our outstanding schema pull request if created with a version
+                    version.ifPresent(v -> {
+                        logger.debug("Successfully processed response to schema pull, removing endpoint from scheduled schema pulls {}: {}", endpoint, v);
+                        MigrationManager.scheduledSchemaPulls.computeIfPresent(v, MigrationManager.removeEndpointFromSchemaPulls(endpoint));
+                    });
```

Comment:
I still don't know if "we don't have multiple parallel requests per node" will be foolproof at the cluster level vs also having some configured backoff duration.

if _this_ node is slow to merge schema, you could still be spamming the other node w/ schema pulls so long as it was respodining quickly enough, no?

Reply:
> Is this node here the local node or the endpoint? The upside would be that we won't make more requests until either we've merged the schema, or our callback has timed out and we need to make a new request anyways.

I see, the MigrationTask#response merges the schema itself before removing, nice!

> Now that we're limiting the number of nodes we hit, in the pathological case:

discussed this offline. this case is possible to hit prior to these changes, and so long as it doesn't take _too_ long to calculate schema in practice the random distribution of requests vs cache expiry should hopefully be ok and not result in indefinite cache misses